lib/statsd.rb

```diff
@@ -369,11 +388,17 @@ def batch(&block)
     Batch.new(self).easy &block
   end
 
+  # Reconnects the socket, useful if the address of the statsd has changed. This
+  # method is not thread safe.
+  def hup
```

Comment:
That sounds like a good suggestion.

An alternative approach would be to open a socket per query instead of keeping it open for the admin piece. While not great it would remove the need for a mutex.